che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -418,8 +419,24 @@ public void clone(CloneRequest request) throws GitException, UnauthorizedExcepti
                 deleteRepositoryFolder();
             }
 
-            throw new GitException(exception.getMessage());
+            String message = generateSSLExceptionMessage(exception);
+            throw new GitException(message, exception);
+        }
+    }
+
+    private String generateSSLExceptionMessage(Throwable e) {
+        String message = e.getMessage();
+
+        //if e caused by an SSLHandshakeException - replace thrown message with a hardcoded message
+        while (e.getCause() != null) {
```

Comment:
`e.getCause()` used three times, may be it would be better to introduce a local variabe